vscode-dotnet-runtime-library/src/Acquisition/IInstallationDirectoryProvider.ts

```diff
@@ -0,0 +1,12 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import { IExtensionState } from '../IExtensionState';
+
+export interface IInstallationDirectoryProvider {
+    getDotnetInstallDir(version: string, installDir: string): string;
```

Comment:
Dotnet is inferred

```suggestion
    getInstallDirectory(version: string, installDir: string): string;
```

Reply:
Why do you need to take in the installDir here? Isn't that the responsibility of this API?

I'd imagine each directory provider would do [this](https://github.com/dotnet/vscode-dotnet-runtime/pull/185/files#diff-62bfd0034cb508600b5a0b8a61f91033d2a9888830f40ff468155779594d958cR31-R32) magic and this API would then only take the `version: string` parameter